core/src/bson_type_info.rs

```diff
@@ -483,6 +483,27 @@ impl BsonTypeInfo {
         simple_type_info: SimpleTypeInfo::default(),
     };
 
+    // Datatype needed for backwards compatibility
+    pub const DATE_OLD: BsonTypeInfo = BsonTypeInfo {
```

Comment:
> When all data types are requested in a call to SQLGetTypeInfo, the result set returned by the function will contain both SQL_TYPE_DATE, SQL_TYPE_TIME, and SQL_TYPE_TIMESTAMP as defined in ODBC 3.x, and SQL_DATE, SQL_TIME, and SQL_TIMESTAMP as defined in ODBC 2.x.

This says we need to return _both_ types when all types are requested. Because we only have type support for SQL_TYPE_TIMESTAMP (93), our "date" type, I added a mirroring ODBC 2 version, which just uses the updated sql types.

Reply:
Yeah I was wondering what we were going to do after reading through the spec. I think this works, but might suggest renaming to 
```suggestion
    pub const LEGACY_DATE: BsonTypeInfo = BsonTypeInfo {
```

Reply:
Though I wonder if we need this here or could just do it in the return? It seems to me that BSON_TYPE_INFO is supposed to represent what the bson types are.